Add some details + ghost cells (#7)
alex-m-h/multipatch-presentation: multi-patchs.tex

\section{Structure for multipatch}
\paragraph{}
We are trying here to answer to the following questions: 
\begin{itemize}

> 	\item What should be done before September? % TO KEEP ? 

Reviewer: Good point I think this does not really belong here, right?

	\item What kind of information do we need to store in each patch? 
	\item What kind of information do the patches need to exchange? 
\end{itemize}

\subsection{Type of global space splitting}
\paragraph{}
The splitting of the global space into patches is currently not fixed. However we can encounter three types of sticking: 
\begin{itemize}

> 	\item simple interface which sticks a whole edge on ne dimension; 
> 	\item T-joint which sticks interfaces with a partial edge; 

Reviewer: ```suggestion
	\item simple interface which sticks two whole edges together; 
	\item T-joint which sticks several edges of to one edge of a patch; 
```

	\includegraphics[width=0.4\textwidth]{images/X_point_in_physical.png}
\caption{X-point in logical and physical domain.}
\end{subfigure}
\caption{\label{Patches_logical_dom} Patches in the logical domain.}
\end{figure}

\newpage

\paragraph{}

> We want here to stay general and to not focus on one patches decomposition of the domain. The figures Fig. \ref{Complex_geometry_1} and \ref{Complex_geometry_2} give example of more complex space. 

Reviewer: ```suggestion
We want to stay general here and not to focus on one specific decomposition of the domain into patches. The figures Fig. \ref{Complex_geometry_1} and \ref{Complex_geometry_2} give examples of more complex spaces. These are not necessarily for the decomposition of the separatrix.
```

\begin{figure*}
	\centering
	\includegraphics*[width=0.8\textwidth]{images/T-joint_sketch.jpg}
	\caption{\label{Complex_geometry_2}
			  Quick sketch of geometry with O-point (purple point) and T-joint (cyan point).
			  There is a T-joint at the cyan point because 
			  edges from patch {\uppercase\expandafter{\romannumeral 2}}
			  and {\uppercase\expandafter{\romannumeral 3}} are connected to the eastern edge
			  of patch {\uppercase\expandafter{\romannumeral 1}}. 
			  Note that the y-dimension of patch {\uppercase\expandafter{\romannumeral 1}}
			  is periodic.}
\end{figure*}

\newpage

\section{Advection equation}
\paragraph{}

> We focus on each part of the equations system solving to answer the questions raised in the introduction. We start with the advection equations. 

Reviewer: ```suggestion
We focus on the advection and Poisson equation separately in the context of the questions raised in the introduction. We start with the advection equation. 
```

\end{aligned}
\end{equation}

> with $A$ the advection field defined on the logical domain on the physical domain axis. Another example is the advection of the drift-kinetic model 

Author: axes (sorry I thought it was one axe, two axis. But it is one axis, two axes).

\begin{equation}
\begin{aligned}
	\partial_t f 
		+ v_{GC}\cdot \nabla_{\perp}f 
		+ v_{\parallel} \partial_z f 
		+ \dot{v}_{\parallel} \partial_{v_{\parallel}} f
		= 0,
\end{aligned}
\end{equation}

> but it seems to raise the same questions as the guiding center equations.

Reviewer: Do you mean that the problems are similar in multipatch geometry?

\end{aligned}
\right.
\end{equation}

> with a time integration method. In the case of Runge-Kutta method, the advection field needs to be evaluated at intermediate feet.

Reviewer: Why did you delete the mention of the discrete flow? $\bar{\psi}$ is still undefined

	
	%\draw[dashed, gray, ->] (1.5, 0.75) -- (2.87, 0.53) node[below]{feet}; 
	%\draw[dashed, gray, <-] (1.5, 1.25) -- (2.87, 1.02) node[above]{value}; 
	
	\node at (2.87, 0.78+0.4) {2.};
	
	
	\begin{scope}[xshift=6cm]
		% physical grid
		\def\RmaxA{1.5}
		\def\RmaxB{2.7}
		\foreach \r in {0, 0.3, ...,\RmaxB} {
		    \draw[lightgray] (0,0) ++(0:\r) arc (0:60:\r);
		}
		\foreach \theta in {0, 15, ..., 60} {
    			\draw[lightgray] (0,0) -- (\theta:\RmaxB);
		}
		\node at (0.75, -0.25) {$1$};
		\node at (2.1, -0.25) {$2$};

		\draw[thick, black] (0,0) ++(0:\RmaxA) arc (0:60:\RmaxA);
		\draw[thick, black] (0,0) ++(0:\RmaxB) arc (0:60:\RmaxB);
		
		\draw[thick, black] (0,0) -- (0:\RmaxB);
		\draw[thick, black] (0,0) -- (60:\RmaxB);
		
		% feet
		\draw[fill = red, color=red]  (1.05,0.58) circle (2pt);
		\node at (1.75, 0.75) {\color{red} $\times$};
		
		\draw[->] (1.05, 0.58-0.25) -- (1.75, 0.75-0.25) node[below right]{1.}; 
		\draw[<-] (1.05, 0.58+0.25) -- (1.75, 0.75+0.25) node[above right]{3.}; 
	\end{scope}
	
	% legend
	\begin{scope}[xshift=10cm]	
		\node[right] at (0,1.5) {1. compute feet.};
		\node[right] at (0,1) {2. evaluate function.};

> 		\node[right] at (0,0.5) {3. transfert value.};

Reviewer: ```suggestion
		\node[right] at (0,0.5) {3. transfer value.};
```

	\end{scope}
\end{tikzpicture}

> \caption{Example of a feet outside the patch 1 in the logical and physical domains.}

Reviewer: A very nice sketch!

> \caption{Example of a feet outside the patch 1 in the logical and physical domains.}

Reviewer: A very nice sketch!

> \caption{Example of a feet outside the patch 1 in the logical and physical domains.}

Author: "characteristic feet" I prefer, but change done.

\end{center}

\paragraph{Further questions on outside feet.}
Problem: the mapping from the logical to the physical domain is not always easily invertible. 

\begin{itemize}

> 	\item For an advection in the physical domain, if the feet is outside of the domain, how to get the feet in the logical domain of another patch (could be a not invertible mapping for the other patch)?

Reviewer: No, the mappings we consider are always invertible almost everywhere with the only exception of O-points. The problem is that the computation is computationally expensive. Your characteristic foot will not land at the O-point when it leaves a patch, I am pretty sure

Author: I think I didnt  pull your changes. I didnt wanted to change this sentence.

	\begin{itemize}
		\item For a given point in the physical domain, how do we know in which subset it belongs to?
	\end{itemize}

> 	\item Let's assume we know in which patch we are, how can we evaluate a function? How can we get $\rho(x,y)$ with $(x,y)$ in the physical domain (especially when the mapping isn't invertible)?

Reviewer: Same as above, the mappings are invertible
